internal/data.go

```diff
@@ -42,11 +42,23 @@ const (
 	JsonEncoding = iota + 1
 )
 
+// CreateNewTimestamp is a method used to construct timestamp from unixNano. In search backend we store internal
+// timestamps as unixNano.
+func CreateNewTimestamp(nano int64) *Timestamp {
+	ts := time.Unix(0, nano).UTC()
+	return &Timestamp{
+		Seconds:     ts.Unix(),
+		Nanoseconds: int64(ts.Nanosecond()),
+		UnixNano:    ts.UnixNano(),
```

Comment:
This looks like a duplication of the above two fields.
We can just have a method `UnixNano() int64 { return t.Seconds * time.Nanosecond + t.Nanoseconds}`

Reply:
sure